README.md
# AMPP-Stack for Vagrant
A nice *Apache, MySQL, PHP, PgSQL*-Stack for Vagrant. Pretty good variety of modules and tools that are pre-installed. Backups your databases (mysql) up on shutdown. Has Git, NodeJS, NPM, Gulp installed and configured. Provides an SSL-VirtualHost for Apache using an automated generated self-signed cert.

Misses nothing, it's your fully-featured dev environment ready in a few seconds.

##  Usage

We support *libvirt* and *virtualbox* as provider!

```
# There's a vvguest installed, but make sure it matches your installation
vagrant plugin install vagrant-vbguest
# Now load and start the image for PHP5.6 stack
vagrant init chrisandchris/amp && vagrant up
# Use the PHP7 stack
vagrant init chrisandchris/amp7 && vagrant up
```

See: https://atlas.hashicorp.com/chrisandchris/boxes/amp and https://atlas.hashicorp.com/chrisandchris/boxes/amp7

## Specification
The boxes consist of:

- Apache 2.4
    - Enabled both HTTP and HTTPS
    - using a self-signed certificate for ssl connections (generated on boot if not yet done)
- PHP from *ppa:ondrej/php5-5.6* with extensions
    - CURL, GD, MCRYPT, MySQL, xDebug, mbstring, PDO
    - xDebug uses idekey `vagrant`
    - extensive error reporting
- *or* PHP 7 when using the *amp7* stack
- xDebug is pre-configured
- Some Alias-Commands:
    - phpx for php with xDebug in CLI
    - phpunitx for phpunit with xDebug in CLI
    - the env variable `HOST` contains the ip of `eth0` (when using private network, the host is there on `.2`
- MySQL 5.5
    - in **strict** mode
        - NO_ZERO_IN_DATE
        - NO_ZERO_DATE
        - ONLY_FULL_GROUP_BY
        - STRICT_ALL_TABLES
    - every database gets automatically backuped into .sql/ directory on `vagrant halt`
        - nice feature for developing wordpress/cms content offline
        - could prevent you from data loss
- PostgreSQL 9.4 with default configuration
    - there is a user `vagrant` having password `vagrant` having access to a database named `vagrant`
- PostgreSQL 9.3 with default configuration
    - there is a user `vagrant` having password `vagrant` having access to a database named `vagrant`
- Git
- NodeJS
- Gulp

## How to improve
Submit an issue or a pull request!


CHANGELOG.md
# Changelog
Both versions, amp7 and amp5.6, are developed simultaneously.

## v8 (only amp stack)
- updated ppa url for php

## v7 (only amp7 stack)
- added missing xDebug configuration for Apache

## v6
- added PgSQL 9.3
- added a default database to both PgSQL 9.3 and 9.4

## v5
- added realpath, locale-gen
- improved bash_profile script insertion

## v4
- added libvirt provider

## v3
- improved xdebug handling, xdebug now gets its ip on its own

## v2
- new release with old stuff, now version number increases together with amp5.6

### v1.0.4
- added mysql backup script
- added self-signed cert for apache ssl (installs on boot if not yet done)

### v1.0.3
- split deps up in before, custom, after
- fixed small issues in scripts
- fixed issues from v1.0.2

### v1.0.2
- has been revoked

### 1.0.1
- xDebug: Disabled profiler by default


test.sh
#!/usr/bin/env bash

command_exists () {
    type "$1" &> /dev/null ;
}

if ! export | grep -Eq '^declare -x ATLAS_TOKEN=' ; then
    echo "You need to export the ATLAS_TOKEN"
    exit 1
fi

declare -a reqs=(
    "packer"
    "vagrant"
    "virtualbox"
    "libvirt"
)

for (( i = 0; i < ${#cmds[@]} ; i++ )); do
    if ! command_exists ${cmds[$i]}; then
        echo "Command not installed, aborting."
        echo ${cmds[$i]}
        exit 1
    fi
done

declare -a cmds=(
    "rm -rf output-virtualbox-iso"
    "packer build -only=virtualbox-iso \"$1.json\""
    "vagrant box add packer_virtualbox-iso_virtualbox.box --name=\"tmp/$1\" --force"
    "rm -rf packer_cache/*"
    "rm -rf output-virtualbox-iso/*"
    "cd test"
    "rm -rf ./.sql_dumps"
    "vagrant destroy -f"
    "vagrant box update"
    "vagrant up $1"
)

echo "Running commands in a row..."

for (( i = 0; i < ${#cmds[@]} ; i++ )); do
    printf "\n**** Running: ${cmds[$i]} *****\n\n"

    if ! eval "${cmds[$i]}"; then
        echo ""
        echo "-------------------------"
        echo "Command failed, aborting."
        echo "${cmds[$i]}"
        exit 1
    fi
done



scripts/dep-7.sh
#!/bin/bash

echo "--- Installing PHP7 and Apache 2.4 ---"
add-apt-repository -y ppa:ondrej/php
apt-get update
apt-get install -y apache2 php7.0 php7.0-dev php7.0-curl php7.0-opcache php7.0-json php7.0-mcrypt php7.0-pgsql php7.0-gd \
    libapache2-mod-php7.0 php7.0-sqlite php-mysql php-mbstring php7.0-pgsql

sed -i "s/;date.timezone =/date.timezone = Europe\/Zurich/" /etc/php/7.0/apache2/php.ini
sed -i "s/;date.timezone =/date.timezone = Europe\/Zurich/" /etc/php/7.0/cli/php.ini
sed -i "s/upload_max_filesize = 2M/upload_max_filesize = 64M/" /etc/php/7.0/apache2/php.ini

echo "--- Manually installing xdebug ---"
wget http://xdebug.org/files/xdebug-2.4.0.tgz /home/vagrant/xdebug-2.4.0.tgz
tar -xvzf xdebug-2.4.0.tgz
cd /home/vagrant/xdebug-2.4.0
phpize
./configure
make
cp modules/xdebug.so /usr/lib/php/20151012
echo "zend_extension = /usr/lib/php/20151012/xdebug.so" >> /etc/php/7.0/cli/php.ini
echo "zend_extension = /usr/lib/php/20151012/xdebug.so" >> /etc/php/7.0/apache2/php.ini
echo "-- Configure xdebug --"
echo "
xdebug.remote_enable=1
xdebug.profiler_enable=0
xdebug.remote_connect_back=1
xdebug.remote_port=9000
xdebug.remote_log=/tmp/xdebug.log
xdebug.max_nesting_level=512" >> /etc/php/7.0/cli/conf.d/20-xdebug.ini
echo "
xdebug.remote_enable=1
xdebug.profiler_enable=0
xdebug.remote_connect_back=1
xdebug.remote_port=9000
xdebug.remote_log=/tmp/xdebug.log
xdebug.max_nesting_level=512" >> /etc/php/7.0/apache2/conf.d/20-xdebug.ini


scripts/dep-5.6.sh
#!/usr/bin/env bash

echo "--- Installing PHP5.6 and Apache 2.4 ---"
add-apt-repository -y ppa:ondrej/php
apt-get update
apt-get install -y php5 apache2 libapache2-mod-php5 php5-curl php5-gd php5-mcrypt \
    php5-mysql php5-dev php5-xdebug php5-pgsql

sed -i "s/;date.timezone =/date.timezone = Europe\/Zurich/" /etc/php5/apache2/php.ini
sed -i "s/;date.timezone =/date.timezone = Europe\/Zurich/" /etc/php5/cli/php.ini
sed -i "s/upload_max_filesize = 2M/upload_max_filesize = 64M/" /etc/php5/apache2/php.ini

echo "-- Configure xdebug --"
echo "
xdebug.remote_enable=1
xdebug.profiler_enable=0
xdebug.remote_connect_back=1
xdebug.remote_port=9000
xdebug.remote_log=/tmp/xdebug.log
xdebug.max_nesting_level=512" >> /etc/php5/apache2/conf.d/20-xdebug.ini
echo "
xdebug.remote_enable=1
xdebug.profiler_enable=0
xdebug.remote_connect_back=1
xdebug.remote_port=9000
xdebug.remote_log=/tmp/xdebug.log
xdebug.max_nesting_level=512" >> /etc/php5/cli/conf.d/20-xdebug.ini


scripts/dep-any.sh
#!/usr/bin/env bash

#!/bin/bash
#
# Setup the the box. This runs as root

echo "--- Let's get to work. Installing now. ---"
locale-gen

echo "--- Updating packages list ---"
apt-get update && apt-get dist-upgrade -y

echo "--- Installing base packages ---"
debconf-set-selections <<< 'mysql-server mysql-server/root_password password root'
debconf-set-selections <<< 'mysql-server mysql-server/root_password_again password root'
apt-get install -y vim curl python-software-properties dpkg-dev wget ca-certificates \
    git-core mysql-server-5.5 python-software-properties software-properties-common htop

curl --silent --location https://deb.nodesource.com/setup_0.12 | bash -
apt-get update
apt-get install --yes nodejs
sudp npm install -g gulp

echo "--- MySQL time ---"
sed -i 's/#max_connections/max_connections/' /etc/mysql/my.cnf
sed -i 's/max_connections[ ]*= 100/max_connections = 1000/' /etc/mysql/my.cnf
sed -i 's/\[mysqld\]/\[mysqld\]\nsql_mode = "NO_ENGINE_SUBSTITUTION,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ONLY_FULL_GROUP_BY"/' /etc/mysql/my.cnf
sed -i "s/bind-address.*=.*127.0.0.1/bind-address=0.0.0.0/g" /etc/mysql/my.cnf
service mysql restart
mysql -uroot -proot << EOF
    USE mysql;
    UPDATE `user` SET Host = '%' WHERE Host = '::1';
    FLUSH PRIVILEGES;
EOF

echo "--- Postgres time ---"
echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
apt-get install
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
apt-get update && apt-get dist-upgrade -y
apt-get -y install postgresql-9.4

echo "-- Configure xdebug --"
echo "xdebug.remote_enable=1
xdebug.profiler_enable=0
xdebug.remote_connect_back=1
xdebug.remote_port=9000
xdebug.remote_log=/tmp/php5-xdebug.log
xdebug.max_nesting_level=512
xdebug.remote_host=10.211.55.2" >> /etc/php5/apache2/conf.d/xdebug.ini

echo "--- Enabling mod-rewrite ---"
a2enmod rewrite

echo "--- Setting up web directory ---"
rm -rf /var/www/html
ln -fs /vagrant /var/www/html
mkdir -p /var/www/uploads
chmod 0777 /var/www/uploads

echo "--- Modify apache user ---"
sed -i "s/export APACHE_RUN_USER=www-data/export APACHE_RUN_USER=vagrant/" /etc/apache2/envvars
sed -i "s/export APACHE_RUN_GROUP=www-data/export APACHE_RUN_GROUP=vagrant/" /etc/apache2/envvars

echo "--- Turn on errors ---"
sed -i "s/error_reporting = .*/error_reporting = E_ALL/" /etc/php5/apache2/php.ini
sed -i "s/display_errors = .*/display_errors = On/" /etc/php5/apache2/php.ini

echo "-- Modify apache configuration --"
sed -i 's/AllowOverride None/AllowOverride All/' /etc/apache2/apache2.conf
sed -i 's/AllowOverride None/AllowOverride All/' /etc/apache2/sites-enabled/000-default.conf
echo "Listen 8000" >> /etc/apache2/ports.conf
cp /etc/apache2/sites-enabled/000-default.conf /etc/apache2/sites-enabled/000-default-8000.conf
sed -i 's/<VirtualHost *:80>/<VirtualHost *:8000>/' /etc/apache2/sites-enabled/000-default-8000.conf

echo "--- Restarting Apache ---"
service apache2 restart

echo "--- Install Composer (PHP package manager) ---"
curl -sS https://getcomposer.org/installer | php
mv composer.phar /usr/local/bin/composer

echo "--- Installing aliases & default config ---"
echo "alias phpunitx=\"./vendor/phpunit/phpunit/phpunit  -dxdebug.remote_host=10.211.55.2 -dxdebug.remote_autostart=1\"" >> /home/vagrant/.bash_profile
echo "alias phpunit=\"./vendor/phpunit/phpunit/phpunit\"" >> /home/vagrant/.bash_profile
echo "alias ll=\"ls -al\"" >> /home/vagrant/.bash_profile
echo "export XDEBUG_CONFIG=\"idekey=vagrant\"" >> /home/vagrant/.bash_profile
echo "EOF
    mkdir -p /vagrant/.sql_dumps
    mysql -uroot -proot -N -e 'show databases;' | while read dbname; do mysqldump -uroot -proot --complete-insert "$dbname" > "/vagrant/.sql_dumps/$dbname".sql; done" >> /home/vagrant/backup_databases.sh
ln -s /home/vagrant/backup_databases.sh /etc/rc0.d/backup_databases.sh
ln -s /home/vagrant/backup_databases.sh /etc/rc6.d/backup_databases.sh

echo "--- Updating again everything, set hostname ---"
apt-get update && apt-get dist-upgrade -y
echo "amp" >> /etc/hostname

echo "--- All done, enjoy! :) ---"


scripts/dep-any-after.sh
#!/usr/bin/env bash

echo "--- Enabling mod-rewrite ---"
a2enmod rewrite

echo "--- Setting up web directory ---"
rm -rf /var/www/html
ln -fs /vagrant /var/www/html
mkdir -p /var/www/uploads
chmod 0777 /var/www/uploads

echo "--- Modify apache user ---"
sed -i "s/export APACHE_RUN_USER=www-data/export APACHE_RUN_USER=vagrant/" /etc/apache2/envvars
sed -i "s/export APACHE_RUN_GROUP=www-data/export APACHE_RUN_GROUP=vagrant/" /etc/apache2/envvars

echo "--- Turn on errors ---"
sed -i "s/error_reporting = .*/error_reporting = E_ALL/" /etc/php5/apache2/php.ini
sed -i "s/display_errors = .*/display_errors = On/" /etc/php5/apache2/php.ini

echo "-- Modify apache configuration --"
sed -i 's/AllowOverride None/AllowOverride All/' /etc/apache2/apache2.conf
sed -i 's/AllowOverride None/AllowOverride All/' /etc/apache2/sites-enabled/000-default.conf
echo "Listen 8000" >> /etc/apache2/ports.conf
cp /etc/apache2/sites-enabled/000-default.conf /etc/apache2/sites-enabled/000-default-8000.conf
sed -i 's/<VirtualHost *:80>/<VirtualHost *:8000>/' /etc/apache2/sites-enabled/000-default-8000.conf

echo "--- Generating and installing ssl certificate cron script ---"
(crontab -l 2>/dev/null; echo "@reboot /home/vagrant/install_cert.sh ") | crontab -
cat <<EOF >> /home/vagrant/install_cert.sh
if [ -f /etc/ssl/vagrant/vagrant.key ]; then
    exit
fi
sudo mkdir -p /etc/ssl/vagrant/
sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
    -keyout /etc/ssl/vagrant/vagrant.key \
    -out /etc/ssl/vagrant/vagrant.crt \
    -subj "/C=CH/ST=Zurich/L=Zurich/O=Your Dev-Env/OU=A vagrant box/CN=172.28.128.3"
sudo a2enmod ssl
cat <<-EOG  > /etc/apache2/sites-enabled/000-default-443.conf
<VirtualHost *:443>
    DocumentRoot /var/www/html
    SSLEngine on
    SSLCertificateFile /etc/ssl/vagrant/vagrant.crt
    SSLCertificateKeyFile /etc/ssl/vagrant/vagrant.key
</VirtualHost>
EOG
sudo service apache2 restart
EOF
chmod +x /home/vagrant/install_cert.sh

echo "--- Restarting Apache ---"
service apache2 restart

echo "--- Install Composer (PHP package manager) ---"
curl -sS https://getcomposer.org/installer | php
mv composer.phar /usr/local/bin/composer

echo "--- Installing aliases, default config, and some tools ---"
# aliases
# sorry for this dirty solution, but escaping that $HOST export is a pain, so i used base64 to cheat me around that
# here's the raw content of the .bash_profile file
#
# export HOST=$(echo `ifconfig eth0 2>/dev/null|awk '/inet addr:/ {print $2}'|sed 's/addr://'` | sed "s/\([0-9]\+\)\.\([0-9]\+\)\.\([0-9]\+\)\.\([0-9]\+\)/\1\.\2\.\3\.2/")
# alias phpx="php -dxdebug.remote_host=\$HOST -dxdebug.remote_autostart=1"
# alias phpunitx="./vendor/phpunit/phpunit/phpunit -dxdebug.remote_host=\$HOST -dxdebug.remote_autostart=1"
# alias phpunit="./vendor/phpunit/phpunit/phpunit"
# alias ll="ls -al"
# export XDEBUG_CONFIG="idekey=vagrant"
#
base64 --decode << EOF >> /home/vagrant/.bash_profile
ZXhwb3J0IEhPU1Q9JChlY2hvIGBpZmNvbmZpZyBldGgwIDI+L2Rldi9udWxsfGF3ayAnL2luZXQg
YWRkcjovIHtwcmludCAkMn0nfHNlZCAncy9hZGRyOi8vJ2AgfCBzZWQgInMvXChbMC05XVwrXClc
LlwoWzAtOV1cK1wpXC5cKFswLTldXCtcKVwuXChbMC05XVwrXCkvXDFcLlwyXC5cM1wuMi8iKQph
bGlhcyBwaHB4PSJwaHAgLWR4ZGVidWcucmVtb3RlX2hvc3Q9XCRIT1NUIC1keGRlYnVnLnJlbW90
ZV9hdXRvc3RhcnQ9MSIKYWxpYXMgcGhwdW5pdHg9Ii4vdmVuZG9yL3BocHVuaXQvcGhwdW5pdC9w
aHB1bml0IC1keGRlYnVnLnJlbW90ZV9ob3N0PVwkSE9TVCAtZHhkZWJ1Zy5yZW1vdGVfYXV0b3N0
YXJ0PTEiCmFsaWFzIHBocHVuaXQ9Ii4vdmVuZG9yL3BocHVuaXQvcGhwdW5pdC9waHB1bml0Igph
bGlhcyBsbD0ibHMgLWFsIgpleHBvcnQgWERFQlVHX0NPTkZJRz0iaWRla2V5PXZhZ3JhbnQi
EOF
# mysql backup script
cat << EOF > /home/vagrant/backup_mysql.sh
#!/usr/bin/env bash

mkdir -p /vagrant/.sql_dumps
mysql -uroot -proot -N -e 'show databases;' | while read dbname; do mysqldump -uroot -proot --databases \$dbname --complete-insert  > /vagrant/.sql_dumps/\${dbname}.sql; done
EOF
chmod +x /home/vagrant/backup_mysql.sh
cat << EOF >> /etc/init/mysql.conf

pre-stop script
    /home/vagrant/backup_mysql.sh
end script

EOF

echo "--- Updating again everything, set hostname, finish some stuff ---"
apt-get update && apt-get dist-upgrade -y
echo "amp" > /etc/hostname
sed -i.old '/^.*packer.*$/d' /etc/hosts
cat << EOF >> /etc/hosts
127.0.0.1   amp
EOF
locale-gen

echo "--- All done, enjoy! :) ---"


scripts/dep-any-before.sh
#!/usr/bin/env bash

echo "--- Let's get to work. Installing now. ---"
locale-gen

echo "--- Updating anything list ---"
apt-get dist-upgrade -y

echo "--- Installing base packages ---"
debconf-set-selections <<< 'mysql-server mysql-server/root_password password root'
debconf-set-selections <<< 'mysql-server mysql-server/root_password_again password root'
apt-get install -y vim curl python-software-properties dpkg-dev wget ca-certificates \
    git-core mysql-server-5.5 python-software-properties software-properties-common htop realpath

curl --silent --location https://deb.nodesource.com/setup_0.12 | bash -
apt-get update
apt-get install --yes nodejs
sudp npm install -g gulp

echo "--- MySQL time ---"
sed -i 's/#max_connections/max_connections/' /etc/mysql/my.cnf
sed -i 's/max_connections[ ]*= 100/max_connections = 1000/' /etc/mysql/my.cnf
sed -i 's/\[mysqld\]/\[mysqld\]\nsql_mode = "NO_ENGINE_SUBSTITUTION,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ONLY_FULL_GROUP_BY"/' /etc/mysql/my.cnf
sed -i "s/bind-address.*=.*127.0.0.1/bind-address=0.0.0.0/g" /etc/mysql/my.cnf
service mysql restart
# allows connections from everywhere
mysql -uroot -proot << EOF
    USE mysql;
    UPDATE user SET Host = '%' WHERE Host = '::1';
    FLUSH PRIVILEGES;
EOF

echo "--- Postgres time ---"
echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
apt-get install
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
apt-get update && apt-get dist-upgrade -y
apt-get -y install postgresql-9.4 postgresql-9.3
sed -i 's/peer/trust/' /etc/postgresql/9.4/main/pg_hba.conf
sed -i 's/peer/trust/' /etc/postgresql/9.3/main/pg_hba.conf
# create database "vagrant" in 9.3
echo "CREATE ROLE vagrant LOGIN ENCRYPTED PASSWORD 'vagrant';" | sudo -u postgres psql -p 5433
sudo -u postgres createdb vagrant --owner vagrant -p 5433
# create database "vagrant" in 9.4
echo "CREATE ROLE vagrant LOGIN ENCRYPTED PASSWORD 'vagrant';" | sudo -u postgres psql -p 5432
sudo -u postgres createdb vagrant --owner vagrant -p 5432


scripts/dep.sh
#!/bin/bash
#
# Setup the the box. This runs as root

echo "--- Let's get to work. Installing now. ---"
locale-gen

echo "--- Updating packages list ---"
apt-get update && apt-get dist-upgrade -y

echo "--- Installing base packages ---"
debconf-set-selections <<< 'mysql-server mysql-server/root_password password root'
debconf-set-selections <<< 'mysql-server mysql-server/root_password_again password root'
apt-get install -y vim curl python-software-properties dpkg-dev wget ca-certificates \
    git-core mysql-server-5.5 python-software-properties software-properties-common htop

curl --silent --location https://deb.nodesource.com/setup_0.12 | bash -
apt-get update
apt-get install --yes nodejs
sudp npm install -g gulp

echo "--- MySQL time ---"
sed -i 's/#max_connections/max_connections/' /etc/mysql/my.cnf
sed -i 's/max_connections[ ]*= 100/max_connections = 1000/' /etc/mysql/my.cnf
sed -i 's/\[mysqld\]/\[mysqld\]\nsql_mode = "NO_ENGINE_SUBSTITUTION,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ONLY_FULL_GROUP_BY"/' /etc/mysql/my.cnf
sed -i "s/bind-address.*=.*127.0.0.1/bind-address=0.0.0.0/g" /etc/mysql/my.cnf
service mysql restart
mysql -uroot -proot << EOF
    USE mysql;
    UPDATE `user` SET Host = '%' WHERE Host = '::1';
    FLUSH PRIVILEGES;
EOF

echo "--- Postgres time ---"
echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
apt-get install
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
apt-get update && apt-get dist-upgrade -y
apt-get -y install postgresql-9.4

echo "--- Installing PHP5.6 and Apache 2.4 ---"
add-apt-repository -y ppa:ondrej/php5-5.6
apt-get update
apt-get install -y php5 apache2 libapache2-mod-php5 php5-curl php5-gd php5-mcrypt \
    php5-mysql php5-dev php5-xdebug php5-apcu

echo "-- Configure xdebug --"
echo "xdebug.remote_enable=1
xdebug.profiler_enable=0
xdebug.remote_connect_back=1
xdebug.remote_port=9000
xdebug.remote_log=/tmp/php5-xdebug.log
xdebug.max_nesting_level=512
xdebug.remote_host=10.211.55.2" >> /etc/php5/apache2/conf.d/xdebug.ini

echo "--- Enabling mod-rewrite ---"
a2enmod rewrite

echo "--- Setting up web directory ---"
rm -rf /var/www/html
ln -fs /vagrant /var/www/html
mkdir -p /var/www/uploads
chmod 0777 /var/www/uploads

echo "--- Modify apache user ---"
sed -i "s/export APACHE_RUN_USER=www-data/export APACHE_RUN_USER=vagrant/" /etc/apache2/envvars
sed -i "s/export APACHE_RUN_GROUP=www-data/export APACHE_RUN_GROUP=vagrant/" /etc/apache2/envvars

echo "--- Turn on errors ---"
sed -i "s/error_reporting = .*/error_reporting = E_ALL/" /etc/php5/apache2/php.ini
sed -i "s/display_errors = .*/display_errors = On/" /etc/php5/apache2/php.ini

echo "-- Modify apache configuration --"
sed -i 's/AllowOverride None/AllowOverride All/' /etc/apache2/apache2.conf
sed -i 's/AllowOverride None/AllowOverride All/' /etc/apache2/sites-enabled/000-default.conf
echo "Listen 8000" >> /etc/apache2/ports.conf
cp /etc/apache2/sites-enabled/000-default.conf /etc/apache2/sites-enabled/000-default-8000.conf
sed -i 's/<VirtualHost *:80>/<VirtualHost *:8000>/' /etc/apache2/sites-enabled/000-default-8000.conf

echo "--- Restarting Apache ---"
service apache2 restart

echo "--- Install Composer (PHP package manager) ---"
curl -sS https://getcomposer.org/installer | php
mv composer.phar /usr/local/bin/composer

echo "--- Installing aliases & default config ---"
etecho "alias phpunitx=\"./vendor/phpunit/phpunit/phpunit  -dxdebug.remote_host=10.211.55.2 -dxdebug.remote_autostart=1\"" >> /home/vagrant/.bash_profile
echo "alias phpunit=\"./vendor/phpunit/phpunit/phpunit\"" >> /home/vagrant/.bash_profile
echo "alias ll=\"ls -al\"" >> /home/vagrant/.bash_profile
echo "export XDEBUG_CONFIG=\"idekey=vagrant\"" >> /home/vagrant/.bash_profile

echo "--- Updating again everything, set hostname ---"
apt-get update && apt-get dist-upgrade -y
echo "amp" >> /etc/hostname

echo "--- All done, enjoy! :) ---"
